src/main/cpp/VideoState.cpp

```diff
@@ -1542,3 +1619,375 @@ void VideoState::sdl_audio_callback(Uint8 *stream, int len) {
 	}
 }
 
+void VideoState::set_speed(const int step) {
+	int i;
+	for (i = 0; i < FF_ARRAY_ELEMS(rate_speed_map); i++) {
+		if (last_speed == rate_speed_map[i].clock_speed) {
+			if (i+step >= 0 && i+step < FF_ARRAY_ELEMS(rate_speed_map)) {
+				if (!isPaused() && rate_speed_map[i + step].clock_speed == 0) {
+					toggle_pause();
+					return; // We can't process the 0 as a new speed for the presentation time
+				}
+				if (!muted && rate_speed_map[i + step].clock_speed != 1) {
+					toggle_mute();
+				}
+				else if (muted && rate_speed_map[i + step].clock_speed == 1) {
+					toggle_mute();
+				}
+#if CONFIG_AVFILTER
+				vfilters = rate_speed_map[i + step].command;
```

Comment:
Do we need locks?

The reader thread presumably accesses these concurrently?

Reply:
Thank you, added a lock for vfilters variable.